### PR TITLE
Rename LineShape2D to WorldMarginShape2D

### PR DESCRIPTION
**Pull request by Calinou**
The new name makes it more obvious that it acts as an infinite plane, and is consistent with its 3D counterpart (WorldMarginShape3D).

See https://github.com/godotengine/godot-proposals/issues/817.

@AndreaCatania I noticed that references to PlaneShape were not changed in the 3D physics server, but I decided to rename them in the 2D physics server. We should make a separate PR for 3D physics server after this one is merged.

### CODE REVIEW

File: servers/physics_2d/collision_solver_2d_sw.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@
 //#define collision_solver gjk_epa_calculate_penetration
 
 bool CollisionSolver2DSW::solve_static_line(const Shape2DSW *p_shape_A, const Transform2D &p_transform_A, const Shape2DSW *p_shape_B, const Transform2D &p_transform_B, CallbackResult p_result_callback, void *p_userdata, bool p_swap_result) {
-	const LineShape2DSW *line = static_cast<const LineShape2DSW *>(p_shape_A);
+	const WorldMarginShape2DSW *line = static_cast<const WorldMarginShape2DSW *>(p_shape_A);
 	if (p_shape_B->get_type() == PhysicsServer2D::SHAPE_LINE) {
```
Comment (pouleyKetchoupp) [inline servers/physics_2d/collision_solver_2d_sw.cpp:39]: In order to keep things consistent in the physics server, we should also rename these:
-`SHAPE_LINE` to `SHAPE_WORLD_MARGIN`
-`solve_static_line()` to `solve_static_world_margin()`
-`line` local variable to `world_margin`

**Review by pouleyKetchoupp — CHANGES_REQUESTED** [review]
Thanks for making this change!

File: servers/physics_3d/shape_3d_sw.h [hunk 2]
```diff
@@ -35,7 +35,7 @@
 #include "servers/physics_server_3d.h"
 /*
 
-SHAPE_LINE, ///< plane:"plane"
+SHAPE_WORLD_MARGIN, ///< plane:"plane"
```
Comment (pouleyKetchoupp) [inline servers/physics_3d/shape_3d_sw.h:38]: Just one last nitpick: this file should only be changed as part of renaming the 3D version :)

**Review by pouleyKetchoupp — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**